crates/ruma-identity-service-api/src/association/bind_3pid/v2.rs

```diff
@@ -0,0 +1,79 @@
+//! [POST /_matrix/identity/v2/3pid/bind](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-3pid-bind)
+
+use std::collections::BTreeMap;
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::{ServerNameBox, ServerSigningKeyId, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Publish an association between a session and a Matrix user ID.",
+        method: POST,
+        name: "bind_3pid",
+        path: "/_matrix/identity/v2/3pid/bind",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// The Session ID generated by the `requestToken` call.
+        pub sid: &'a str,
+
+        /// The client secret passed to the `requestToken` call.
+        pub client_secret: &'a str,
+
+        /// The Matrix user ID to associate with the 3PIDs.
+        pub mxid: UserId,
```

Comment:
```suggestion
        pub mxid: &'a UserId,
```

https://github.com/ruma/ruma/blob/main/CONTRIBUTING.md#borrowing-request-types